File: app/src/main/java/com/blues/module_main/MainPageFragment.java
package com.blues.module_main;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import com.blues.R;
import com.blues.framework.base.BaseFragment;
import com.blues.model_wanandroid.WanAndroidBannerEntity;
import com.blues.model_wanandroid.WanAndroidEntity;
import com.blues.model_wanandroid.WanAndroidViewHolder;
import com.jude.easyrecyclerview.EasyRecyclerView;
import com.jude.easyrecyclerview.adapter.BaseViewHolder;
import com.jude.easyrecyclerview.adapter.RecyclerArrayAdapter;

import androidx.recyclerview.widget.LinearLayoutManager;

/**
 * User : Blues
 * Date : 2019/4/11
 * Time : 15:27
 */

public class MainPageFragment extends BaseFragment implements BannerContract.iBannerView {

    private MainPageHeaderView mHeader;
    private BannerContract.iBannerPresenter iPresenter;

    @Override
    public int setLayoutResourceId() {
        return R.layout.fragment_mainpage;
    }

    @Override
    public void initLayout(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {

        mPresenter = new BannerPresenter(this);

        iPresenter.getBanner();

        EasyRecyclerView ervMain = rootView.findViewById(R.id.erv_main);
        ervMain.setLayoutManager(new LinearLayoutManager(mContext));
        RecyclerArrayAdapter<WanAndroidEntity.DataBean.DatasBean> mAdapter;
        ervMain.setAdapter(mAdapter = new RecyclerArrayAdapter<WanAndroidEntity.DataBean.DatasBean>(mContext) {
            @Override
            public BaseViewHolder OnCreateViewHolder(ViewGroup parent, int viewType) {
                return new WanAndroidViewHolder(parent);
            }
        });
        mAdapter.addHeader(mHeader = new MainPageHeaderView(mContext));
    }


    @Override
    public void lazyFetchData() {
    }

    @Override
    public void onPause() {
        super.onPause();
        mHeader.onBannerPause();
    }

    @Override
    public void onResume() {
        super.onResume();
        mHeader.onBannerResume();
    }

    @Override
    public void showRequestError(String msg) {

    }

    @Override
    public void onBannerReqSuccess(WanAndroidBannerEntity banner) {
        if (banner != null) {
            mHeader.initBanner(mContext, banner);
        }
    }

    @Override
    public void onBannerReqFailure(String result) {

    }

    @Override
    public void setPresenter(BannerContract.iBannerPresenter presenter) {
        iPresenter = presenter;
    }
}

File: app/src/main/java/com/blues/module_main/view/MainPageFragment.kt
package com.blues.module_main.view

import com.blues.R
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import com.jude.easyrecyclerview.EasyRecyclerView
import androidx.recyclerview.widget.LinearLayoutManager
import com.jude.easyrecyclerview.adapter.RecyclerArrayAdapter
import com.blues.model_wanandroid.model.WanAndroidEntity.DataBean.DatasBean
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.blues.framework.base.BaseKoinFragment
import com.blues.model_wanandroid.view.WanAndroidViewHolder
import com.blues.module_main.model.WanAndroidBannerEntity
import com.blues.module_main.vm.WanAndroidBannerViewModel
import com.jude.easyrecyclerview.adapter.BaseViewHolder
import org.koin.androidx.viewmodel.ext.android.viewModel

/**
 * User : Blues
 * Date : 2019/4/11
 * Time : 15:27
 */
class MainPageFragment : BaseKoinFragment() {

    private val bannerViewModel: WanAndroidBannerViewModel by viewModel()

    private var mHeader: MainPageHeaderView? = null

    override fun getLayoutId(): Int = R.layout.fragment_mainpage

    override fun initData(inflater: LayoutInflater, container: ViewGroup?, saveInstanced: Bundle?) {
        bannerViewModel.getBanner()

        rootView.apply {
            val ervMain: EasyRecyclerView = findViewById(R.id.erv_main)
            var mAdapter: RecyclerArrayAdapter<DatasBean?>

            ervMain.apply {
                setLayoutManager(LinearLayoutManager(requireActivity()))
                adapter = object : RecyclerArrayAdapter<DatasBean?>(requireActivity()) {
                    override fun OnCreateViewHolder(parent: ViewGroup,
                        viewType: Int): BaseViewHolder<*> {
                        return WanAndroidViewHolder(parent)
                    }
                }.also { mAdapter = it }
            }
            mAdapter.addHeader(MainPageHeaderView(requireActivity()).also { mHeader = it })

            bannerViewModel.wanAndroidBanner.observe(requireActivity(), { wanAndroidBannerEntity: WanAndroidBannerEntity? ->
                run {
                    wanAndroidBannerEntity?.let {
                        mHeader?.initBanner(requireActivity(), it)
                    }
                }
            })
        }
    }
}